chat-android/src/main/java/com/ably/chat/Message.kt

```diff
@@ -53,7 +56,7 @@ data class Message(
      * Do not use metadata for authoritative information. There is no server-side
      * validation. When reading the metadata treat it like user input.
      */
-    val metadata: MessageMetadata?,
+    val metadata: MessageMetadata,
```

Comment:
Found `metadata` is non-nullable in realtime/rest response

Reply:
Integration test and assertions for the same in [MessagesIntegrationTest.kt](https://github.com/ably/ably-chat-kotlin/blob/be433f2577bddf1172e26dbfe8fda77cb434c1a7/chat-android/src/test/java/com/ably/chat/integration/MessagesIntegrationTest.kt#L78)

Reply:
There’s nothing in the spec that says `Metadata` is non-null. Andy mentioned it’s nullable. I think we should treat it as nullable and `?: JsonObject()` in 2 places when we create Message object (Messages and ChatApi)

Reply:
<img width="787" alt="image" src="https://github.com/user-attachments/assets/c7f04630-62f3-466c-9e90-dce3af56013e" />

Reply:
As I mentioned, since response is `non-nullable` creating `?: JsonObject()` is redudunt for the given code. 
Even if andy mentioned it's nullable, I didnt observe the same in the response. So, if we have a possibility in the future that it can become nullable, we should apply same logic for both `headers` and `metadata`.

Reply:
So going with option 2 as per discussion https://github.com/ably/ably-chat-kotlin/pull/100/files#r1969694983.
Enforced compiler level defaults when marked both fields as `non-nullable`.